### PR TITLE
Update deps for 27

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses "compile on Erlang 27".

Proposed changes include:
* Update amqp_client and rabbit_common
* Use rabbitmq_common with a fix https://github.com/pma/amqp/issues/231
* Update luerl to fix compilation error (dependency of nkservice)
* Update worker pool. Override minimum_otp_vsn for worker_pool

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/232243) / elasticsearch_and_cassandra_mnesia / dcfee1f88ab34f4ac5a257a6f773ccaa2f409d50
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4338/232243/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4338/232243/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40497d2fc1ebfd.2024-07-25_16.04.35/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

File: rebar.config [hunk 1]
```diff
@@ -217,7 +221,8 @@
    {override, jwerl, [{plugins, [rebar3_elixir, rebar3_hex]}]},
    %% GraphQL pulls geas_rebar3, which is a dev dependency
    %% geas takes some time to compile, but we do not need it
-   {override, graphql, [{plugins, [rebar3_hex]}]}
+   {override, graphql, [{plugins, [rebar3_hex]}]},
```
Comment (NelsonVides) [inline rebar.config:224]: Btw, can we fix this directly on the dep? https://github.com/esl/graphql-erlang? We have our own fork so it's easy to fix there.

File: rebar.config [hunk 2]
```diff
@@ -118,7 +119,10 @@
   {jwerl, "1.2.0"},
   {cpool, "0.1.0"}, %% Do not upgrade cpool to version 0.1.1, it has bugs
   {nkpacket, {git, "https://github.com/esl/nkpacket.git", {branch, "mongooseim-ranch-compatibility"}}},
-  {nksip, {git, "https://github.com/arcusfelis/nksip.git", {branch, "mu-fix-dialyzer"}}}
+  {nksip, {git, "https://github.com/arcusfelis/nksip.git", {branch, "mu-fix-dialyzer"}}},
```
Comment (NelsonVides) [inline rebar.config:122]: I wish we could just get rid of these nk dependencies to begin with...

**Review by chrzaszcz — APPROVED** [review]
Looks good 👍

File: rebar.config (hunk 1 above)
Comment (arcusfelis) [reply rebar.config:224]: PR created https://github.com/esl/graphql-erlang/pull/5

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/233522) / elasticsearch_and_cassandra_mnesia / 4e5a085bf406979110f742ddec549c5dc6878371
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4338/233522/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4338/233522/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40b324e2b797fa.2024-07-29_20.55.07/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

**Review by NelsonVides — APPROVED** [review]

**Pull request merged**